upstairs/src/lib.rs

```diff
@@ -498,18 +498,53 @@ async fn process_message(
             )
         }
         Message::ErrorReport {
-            upstairs_id: _,
-            session_id: _,
+            upstairs_id,
+            session_id,
             job_id,
             error,
         } => {
-            // XXX currently, this error report goes nowhere except to the log.
-            // The Upstairs should track this for each Downstairs.
+            // The Upstairs should not consider a job completed until it has
+            // returned an Ok result, and should therefore log and eat all
+            // ErrorReport messages here. This will change in the future when
+            // the Upstairs tracks the number of errors per Downstairs, and acts
+            // on that information somehow.
             error!(
                 u.log,
                 "[{}] job id {} saw error {:?}", client_id, job_id, error
             );
-            return Ok(());
+
+            // However, there is one case (see `check_message_for_abort` in
+            // downstairs/src/lib.rs) where the Upstairs **does** need to
+            // act: when a repair job in the Downstairs fails, that Downstairs
+            // aborts itself and reconnects.
+            let _active = u.active.lock().await;
+            let ds = u.downstairs.lock().await;
+
+            if let Some(job) = ds.ds_active.get(job_id) {
+                if job.work.is_repair() {
+                    // Return the is error and let the previously written error
+                    // processing code work.
+                    cdt::ds__repair__done!(|| (job_id, client_id as u64));
+
+                    // XXX uncomment this to see the upstairs disconnect this
```

Comment:
We should take out the XXX comments here, I think once we know this does what we expect
we don't need to comments.  I don't know why (other than testing) we would want to get the
upstairs stuck in live repair.

Reply:
I'm allowed a few XXX, this PR's still in draft! :P